pygit2/remote.py

```diff
@@ -91,12 +91,30 @@ def push_url(self):
 
         return maybe_string(C.git_remote_pushurl(self._remote))
 
-    def connect(self, callbacks=None, direction=C.GIT_DIRECTION_FETCH):
+    def connect(self, callbacks=None, direction=C.GIT_DIRECTION_FETCH, proxy=None):
         """Connect to the remote.
+
+        Parameters:
+
+        proxy : None or True or str
+            Proxy configuration. Can be one of:
+
+            * `None` (the default) to disable proxy usage
+            * `True` to enable automatic proxy detection
+            * an url to a proxy (`http://proxy.example.org:3128/`)
         """
+        if proxy is not None:
+            proxy_opts = ffi.new('git_proxy_options *')
+            if proxy is True:
+                proxy_opts.type = C.GIT_PROXY_AUTO
+            elif proxy is not None:
+                proxy_opts.type = C.GIT_PROXY_SPECIFIED
+                proxy_opts.url = proxy
```

Comment:
This shouldn't work because proxy will be a text string and we have to pass bytes to the C library.
You can do this: `proxy_opts.url = to_bytes(proxy)`

Also we should raise `TypeError` if `proxy`is of an unexpected type: something else than `None`, boolean or string.

And since the same code is repeated 3 times, it would be better to have a helper function, something like `set_proxy(proxy_opts, proxy)`

Reply:
@jdavid Done. Please review again.